src/decimal.rs

```diff
@@ -1097,27 +1103,33 @@ fn sub_internal(value: &mut [u32], by: &[u32]) -> u32 {
     overflow
 }
 
+//#[inline]
 fn sub_part(left: u32, right: u32, overflow: u32) -> (u32, u32) {
-    let mut invert = false;
-    let overflow = i64::from(overflow);
-    let mut part: i64 = i64::from(left) - i64::from(right);
-    if left < right {
-        invert = true;
-    }
-
-    if part > overflow {
-        part -= overflow;
-    } else {
-        part -= overflow;
-        invert = true;
-    }
+    let part = 0x1_0000_0000u64 + u64::from(left) - (u64::from(right) + u64::from(overflow));
```

Comment:
I'd probably feel a little better here with brackets around the first section to make the order of operations very blatantly obvious (i.e. `(0x1_0000_0000u64 + u64::from(left))`). 

In addition, to clean up it'd be a good idea to remove the superfluous comments below. We'll still have a copy of how it worked within Git if we need to revert.

I guess the last suggestion to round this off a bit further; it'd be great to ensure we have a test (or tests) that covers an extreme scenario at a part level. e.g. maybe 0XFFFF_FFFF_FFFF_FFFF - -0XFFFF_FFFF_FFFF_FFFF_FFFA or something like that. I've been trying a couple of examples on paper to test the logic but another addition to the test array would be ideal. 😄

Reply:
Done